apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/index.js

```diff
@@ -415,3 +420,14 @@ registerPlugin( 'page-templates-sidebar', {
 		);
 	},
 } );
+
+// Make sidebar plugin open by default.
+if (
+	! wpSelect( 'core/edit-post' ).isEditorPanelOpened(
+		'page-templates-sidebar/Template Modal Opener'
+	)
+) {
+	wpDispatch( 'core/edit-post' ).toggleEditorPanelOpened(
+		'page-templates-sidebar/Template Modal Opener'
+	);
+}
```

Comment:
I didn't experience the thing Andy did locally, but I think we could solve the problem by making it a bit more robust. I think it's possible that at this point in execution, the state hasn't fully loaded. What about this? Alternatively, maybe it would work on `domReady`

```suggestion
import { subscribe } from '@wordpress/data';

const unsubscribe = subscribe( () => {
  if (
	  ! wpSelect( 'core/edit-post' ).isEditorPanelOpened(
		'page-templates-sidebar/Template Modal Opener'
	  )
  ) {
	  wpDispatch( 'core/edit-post' ).toggleEditorPanelOpened(
		'page-templates-sidebar/Template Modal Opener'
	  );
	  // Only open it once:
	  unsubscribe();
  }
} );
```

Reply:
I agree with @noahtallen.
Also, we have used the `subscribe` trick in other occasions in FSE, so it all sounds good to me. 👍

Reply:
Great recommendation, thanks!  I updated it to use the subscribe funciton.

Reply:
It works great, but it is not what I expected from `subscribe`.  That is, `subscribe` is documented as running whenever the state changes.  But when we open or close the modal, isn't that updating the state?  If that was the case I would expect this check to run every time we minimize it in the sidebar, which would force it to open again?  But with testing that is not how it works, so I am a little confused on that.

Reply:
The `unsubscribe()` that is called when the panel is opened is exactly the reason why it doesn't happen!

Thinking in normal JS terms, `subscribe/unsubscribe` are similar to `setInterval/clearInterval`:
```js
const waitForOpenPanel = setInterval( () => {
  if ( isPanelOpen ) {
    clearInterval( waitForOpenPanel );
  }
} );
```

Reply:
ooohhhh... I see.   So in this case we are unsubscribing as soon as it runs, but we are subscribing in the first place to ensure that state has been fully loaded and everything is established before checking and dispatching to it?

Reply:
right on!